rdrf/rdrf/views/lookup_views.py

```diff
@@ -1,23 +1,20 @@
-from django.http import HttpResponse
+import logging
+
+from django.http import JsonResponse
 from django.views.generic import View
 from django.urls import reverse
-from django.utils.decorators import method_decorator
-from django.contrib.auth.decorators import login_required
-from django.core.exceptions import PermissionDenied
-
-import json
 
 from registry.patients.models import Patient
 
 from rdrf.helpers.registry_features import RegistryFeatures
+from rdrf.security.mixins import StaffMemberRequiredMixin
+
 
-import logging
 logger = logging.getLogger(__name__)
 
 
-class PatientLookup(View):
+class PatientLookup(StaffMemberRequiredMixin, View):
```

Comment:
For this one, I'm not sure if requiring staff is the best option.

Can't we just use the `PatientManager.get_by_user_and_registry` method as a base query to make the filter by user consistent with other parts of the system?

While we're at it you can also replace the `try/catch` with a `get_object_or_404`.

Reply:
Updated.